### ISSUE TITLE
[css-transforms] UA stylesheet rule to give SVG elements a default transform-box value of view-box

### ISSUE THREAD

**Issue by smfr** [w3c/csswg-drafts#928]
https://www.w3.org/mid/op.x5otn9g7fwt8au@localhost.localdomain

CSS WG resolved to specify a UA stylesheet rule for SVG making the default transform-box for SVG elements be the view-box.

**Comment by smfr** [comment]
svg|* { transform-box: view-box; }

**Comment by smfr** [comment]
Even with a corrected UA rule, we still need prose about the used value to handle the case where the author set the transform-box to border-box.

Want to also have something that can be used at https://drafts.fxtf.org/fill-stroke/#fill-origin

**Comment by smfr** [comment]
svg|* > svg|* {transform-box: view-box}

**Comment by AmeliaBR** [comment]
We might as well follow the syntax in the [user agent stylesheet from SVG 2](https://www.w3.org/TR/SVG2/styling.html#UAStyleSheet), which sets default values of `transform-origin`.  (Note that that user agent stylesheet uses a default namespace rule).

`svg|*:not(svg), svg|*:not(foreignObject) > svg {transform-box: view-box}`

The important net difference from the above is that SVG elements that are direct children of foreignObject elements are excluded.

In other words, SVG namespaced elements are set to `transform-box: view-box` _except_ for the elements that have CSS layout  boxes:

- root `<svg>` elements
- inline SVG elements that are direct children of non-SVG elements
- an `<svg>` element that is the direct child of a `<foreignObject>`

**Comment by AmeliaBR** [comment]
Per the resolutions referenced in https://github.com/w3c/csswg-drafts/issues/857#issuecomment-296024182, it looks like no new user-agent rules will be required.  The different behavior between elements in SVG and CSS layout is going to be handled by the mapping of SVG boxes to CSS boxes.

**Comment by smfr** [comment]
Closing per above comment (now issue #999).
